app/transformers/DeclarationTransformer.scala

```diff
@@ -147,12 +147,15 @@ class DeclarationTransformer {
   }
 
   private def removeAdditionalShareAssetField(json: JsValue): Reads[JsObject] = {
-    val updatedArray: JsArray = json.transform(pathToShareAssets.json.pick[JsArray]) match {
-      case JsSuccess(array, _) => JsArray(array.value.map(x => x.as[JsObject] - IS_PORTFOLIO))
-      case _ => JsArray()
+    if (json.transform((pathToShareAssets).json.pick).isSuccess) {
```

Comment:
Could this be refactored to something like
```
json.transform(pathToShareAssets.json.pick[JsArray]) match {
    case JsSuccess(array, _) =>
        val updatedArray: JsArray = JsArray(array.value.map(x => x.as[JsObject] - IS_PORTFOLIO))
        prunePathAndPutNewValue(pathToShareAssets, updatedArray)
    case _ =>
        doNothing()
}
```

Reply:
I'll give it a try, that looks nicer

Reply:
Updated. Not sure why I didn't do that in the first place, but whatever